docker/config/config.go

```diff
@@ -28,12 +40,21 @@ func (c *Config) IsEmpty() bool {
 func (c *Config) GetCredentials(registry string) (string, string, bool) {
 	_, defined := c.usernames[registry]
 	if !defined {
-		return "", "", false
+		return DefaultUsername, DefaultUsername, false
 	}
 
 	return c.usernames[registry], c.passwords[registry], true
 }
 
+// GetRegistryAuth gets per-registry base64 authentication string
+func (c *Config) GetRegistryAuth(registry string) (string, bool) {
+	username, password, defined := c.GetCredentials(registry)
+
+	jsonString := fmt.Sprintf("{ \"username\": \"%s\", \"password\": \"%s\" }", username, password)
```

Comment:
maybe you can use `{ "username": "%s", "password": "%s" }` to avoid some slashes?

Reply:
ups sorry @Fsero but I can not get what do you mean :cry

Reply:
I need to merge this so one thing I need to push for nois macos would be unblocked.